### PR TITLE
Komarch 12 backpack

### PR DESCRIPTION
**Pull request by yanchith**
This adds the backpack admin interface. For now, it just manages the user table.

Pretty much all of the files in this diff are generated by backpack.

The PR is draft, because I still need to request the noncommercial license for backpack, and we probably shouldn't deploy without it.

### CODE REVIEW

**Review by igor-kamil — COMMENTED** [review]
you can find some inspiration in:
https://github.com/SlovakNationalGallery/register-architektury/pull/8
https://github.com/SlovakNationalGallery/register-architektury/pull/17/files

**Comment by yanchith** [comment]
@igor-kamil Thank you for the suggestion. Assets in `public/packages` are now ignored by git and generated by a hook after `composer install`. There's still quite a few generated files in this PR, but should be much less now.

File: app/Http/Controllers/Admin/UserCrudController.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+     * @see  https://backpackforlaravel.com/docs/crud-operation-list-entries
+     * @return void
+     */
+    protected function setupListOperation()
+    {
+        CRUD::column('name');
+        CRUD::column('email');
+        CRUD::column('email_verified_at');
+        CRUD::column('password');
+        CRUD::column('remember_token');
```
Comment (igor-kamil) [inline app/Http/Controllers/Admin/UserCrudController.php:46]:
```suggestion
```
please hide them

File: app/Http/Controllers/Admin/UserCrudController.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+     */
+    protected function setupCreateOperation()
+    {
+        CRUD::setValidation(UserRequest::class);
+
+        CRUD::field('name');
+        CRUD::field('email');
+        CRUD::field('email_verified_at');
+        CRUD::field('password');
+        CRUD::field('remember_token');
```
Comment (igor-kamil) [inline app/Http/Controllers/Admin/UserCrudController.php:71]:
```suggestion
        CRUD::field('name');
        CRUD::field('email');
        CRUD::field('password');
```
remove the fields, which shouldn't be editable by user + add `password_confirmation` instead
also don't forget to hash password when storing it. get inspired by https://github.com/Laravel-Backpack/PermissionManager/blob/a21231b4ca1bce0199e778aea21109043f4c6728/src/app/Http/Controllers/UserCrudController.php#L129-L141 
(btw. you can get inspired by https://github.com/Laravel-Backpack/PermissionManager/blob/master/src/app/Http/Controllers/UserCrudController.php in general -> also check the validation rules in https://github.com/Laravel-Backpack/PermissionManager/blob/master/src/app/Http/Requests/UserUpdateCrudRequest.php and https://github.com/Laravel-Backpack/PermissionManager/blob/master/src/app/Http/Requests/UserStoreCrudRequest.php )

**Comment by eronisko** [comment]
Sorry for butting in, just a side-note: I wouldn't track (in git) any "untouched" code generated by backpack. Most of this stuff works such that you can `publish` and then simply remove everything you're not touching. It'll still get used as-is from the original repo.

**Comment by yanchith** [comment]
> Sorry for butting in, just a side-note: I wouldn't track (in git) any "untouched" code generated by backpack. Most of this stuff works such that you can `publish` and then simply remove everything you're not touching. It'll still get used as-is from the original repo.

@eronisko Could you say what exactly you mean by that? The static assets are already ignored in git. There's a few generated php files, but is there a strategy to determine which is safe to include in version control and which not? Also how would you exclude them? A blacklist in git-ignore? Does backpack's code generator detect if the file is already there and modified, and not overwrite it?

**Comment by yanchith** [comment]
@igor-kamil I addressed your concerns (inspired by the links you sent). Some parts look very concerning to me (like detecting where in the URL is the user id), but maybe I just need to get used to it.

Please have another look - the changes are in the last commit.

**Comment by eronisko** [comment]
> @eronisko Could you say what exactly you mean by that? The static assets are already ignored in git. There's a few generated php files, but is there a strategy to determine which is safe to include in version control and which not?

I guess I was talking about this: https://backpackforlaravel.com/docs/4.1/base-about#published-views 

Simply put, if there's a file in `/resources/views/vendor/backpack`, it'll use that instead of its original counterpart in `/vendor/backpack`. So if we're not modifying the file (in  `/resources/views/vendor/backpack`), then there's no need to keep our own copy -- less code in our repo.

**Comment by yanchith** [comment]
> Simply put, if there's a file in /resources/views/vendor/backpack, it'll use that instead of its original counterpart in /vendor/backpack. So if we're not modifying the file (in /resources/views/vendor/backpack), then there's no need to keep our own copy -- less code in our repo.

So I tried it, but backpack seems to have generated code specific to our app in those published views. If we fall back to the generic views, our specific changes disappear (e.g. link to user management section). I might be missing something though.

Overall, there's just three files in `/resources/views/vendor/backpack` and they only contribute 16 lines, so I am not terribly sure it is worth pursuing their removal.

**Review by igor-kamil — APPROVED** [review]
now it's ok. thanx!

**Pull request merged**